ref(profiling): remove support for metrics with profile namespace

viglia opened the pull request:
As we moved away from `generic-metrics` we won't need to support metrics with profile namespace anymore.

#skip-changelog

Commit:
remove support for metrics with profile namespace

Commit:
update changelog

Commit:
fix test

Commit:
deprecate profiles_function_generic_metrics_enabled option

jjbayer commented on CHANGELOG.md:
```diff
@@ -12,6 +12,10 @@
 
+**Internal**:
+
+- Remove support for metrics with profile namespace. ([4391](https://github.com/getsentry/relay/pull/4391))
```
```suggestion
- Remove support for metrics with profile namespace. ([#4391](https://github.com/getsentry/relay/pull/4391))
```

jjbayer commented on relay-server/src/services/store.rs:
```diff
@@ -663,17 +663,6 @@ impl StoreService {
-                    .options
-                    .profiles_function_generic_metrics_enabled
-                {
-                    return Ok(());
```
Don't we actually want to always drop profile metrics now? I.e. should we keep the namespace as `Deprecated1` o.a. and always `return Ok(())` for it?

jjbayer commented on tests/integration/test_metrics.py:
```diff
@@ -1905,44 +1905,6 @@ def test_missing_global_filters_enables_metric_extraction(
-    received_metrics = metrics_without_keys(
-        json.loads(metrics_item.get_bytes().decode()), keys={"metadata"}
-    )
-    assert received_metrics == [
```
Related to the question above, what would `received_metrics` be now that the namespace has been removed?

Commit:
fix readme

Commit:
re-introduce integration test

This tests that support for profile metrics is indeed removed by
checking that minisentry.captured_events returns an emprt envelope
